adagrams/game.py

```diff
@@ -1,11 +1,127 @@
+import random
+
 def draw_letters():
-    pass
+
+    LETTER_POOL = {
+        'A': 9, 
+        'B': 2, 
+        'C': 2, 
+        'D': 4, 
+        'E': 12, 
+        'F': 2, 
+        'G': 3, 
+        'H': 2, 
+        'I': 9, 
+        'J': 1, 
+        'K': 1, 
+        'L': 4, 
+        'M': 2, 
+        'N': 6, 
+        'O': 8, 
+        'P': 2, 
+        'Q': 1, 
+        'R': 6, 
+        'S': 4, 
+        'T': 6, 
+        'U': 4, 
+        'V': 2, 
+        'W': 2, 
+        'X': 1, 
+        'Y': 2, 
+        'Z': 1
+    }
+    
+    # initiate draw_letters to record and return drawn random letters
+    # initiate letter_freq to check the frequency of drawn out random letters
+    draw_letters_list = []
+    letter_freq = {}
+    
+    # create random letter picking from the letter poll
+    # only append letters that are no bigger than letter poll frequency
+    while len(draw_letters_list) < 10:
+        cur_random_letter = random.choice(list(LETTER_POOL.keys()))
```

Comment:
Notice that `list(LETTER_POOL.keys()` returns a list with 26 letters A-Z. This doesn't quite reflect the distribution of letters that is represented in the LETTER_POOL dictionary. How could you iterate over the LETTER_POOL dictionary to create a list with 9 As, 2 Bs, 2 Cs, etc..?

Also, nice job using the built-in method choice(). How does choice work under the hood? How would your own implementation of randomly selecting a letter from a list look? 

I encourage students to practice writing this kind of logic on their own because it provides practice in thinking through the algorithm to grab a random letter and provides the muscle memory of writing loops and using random too. While writing it out on your own is more lines of code, I think that's ok for initial projects to you get this practice.

Especially in interviews, we should feel confident that we understand how built-in functions work under the hood because your interviewer may ask you to explain it to them.